src/porepy/models/constitutive_laws.py

```diff
@@ -2462,22 +2495,80 @@ def matrix_porosity(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
             Cell-wise porosity operator [-].
 
         """
+
+        # Sanity check
         if not all([sd.dim == self.nd for sd in subdomains]):
             raise ValueError("Subdomains must be of dimension nd.")
+
+        # Add contributions to poromechanics porosity
+        phi = (
+            self.reference_porosity(subdomains)
+            + self.pressure_contribution(subdomains)
+            + self.displacement_divergence_contribution(subdomains)
```

Comment:
Very good. I think we need to rename methods to porosity_change_from_displacement or similar, though. I can explain the reasoning in person.